### PR TITLE
Update scenario_config.csv for SSP3

### PR DESCRIPTION
**Pull request by laurinks**
## Purpose of this PR

Update `scenario_config.csv` for SSP3 and add SSP2-rollBack:
- Set `cm_prtpScen = 3` for SSP3 scenarios
- Add `SSP3-NPi_impCM-calibrate` for calibration with imperfect capital markets
- For `SSP3-rollBack` and `SSP2-rollBack`, set `carbonprice = none` together with `cm_emiscen = 9` for carbon prices equal to zero from `cm_startyear` onward, set `techpol = none` to remove NPi technology policies from `cm_startyear` onward
- For `SSP3-rollBack`, set `c_techAssumptScen = 4` for SSP3 settings

## Type of change

- [x] Activating new default scenarios (SSP3)

## Checklist:

- [x] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [x] I performed a self-review of my own code
- [x] I explained my changes within the PR, particularly in hard-to-understand areas
- [ ] I checked that the [in-code documentation](https://github.com/remindmodel/remind/blob/develop/main.gms#L120) is up-to-date
- [ ] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [ ] I adjusted `forbiddenColumnNames` in [readCheckScenarioConfig.R](https://github.com/remindmodel/remind/blob/develop/scripts/start/readCheckScenarioConfig.R) in case the PR leads to deprecated switches
- [x] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [ ] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

### CODE REVIEW

File: config/scenario_config.csv
Comment (laurinks) [inline config/scenario_config.csv:1]:
```
File comparison: /p/projects/rd3mod/github/repos/remindmodel/remind/develop/config/scenario_config.csv -> ./config/scenario_config.csv
Columns deleted: -
Columns added:   capitalMarket, cm_prtpScen
Renamed columns: -
Renamed rows:    -

Changes in the scenarios:
+ SSP2-rollBack was added.
~ SSP3-NPi-calibrate:
    cm_prtpScen: NA -> 3 (default: 1)
+ SSP3-NPi_impCM-calibrate was added.
~ SSP3-NDC:
    start: 0 -> 1
    cm_prtpScen: NA -> 3 (default: 1)
~ SSP3-NPi2025:
    start: 0 -> 1
    cm_prtpScen: NA -> 3 (default: 1)
~ SSP3-PkBudg1000:
    start: 0 -> 1
    cm_prtpScen: NA -> 3 (default: 1)
~ SSP3-rollBack:
    start: 0 -> 1
    c_techAssumptScen: 1 -> 4 (default: 1)
    cm_prtpScen: NA -> 3 (default: 1)

$allwarnings
NULL
```

**Comment by laurinks** [comment]
@flohump : Should be set `cfg_mag$gms$s15_elastic_demand <- 1` also in our default coupling config file? Or was this a ScenarioMIP specific thing that should not be used as the default? Thanks!

**Comment by nicobauer** [comment]
Hi Laurin, I am a bit puzzled. we have a `scenario_config_SSP3` that looks like a `*coupled*.csv` and there is no actual `scenario_config_SSP3.csv`. In my understanding there always needs to be a `scenario_config_XXX.csv` and a corresponding `scenario_config_XXX_coupled.csv`.
Maybe I missed something along the way or my routines are too strict and old-fashioned.

**Comment by laurinks** [comment]
Hi Nico, thanks for asking. My understanding from our Mattermost conversation was that we update the SSP3 scenarios in `scenario_config.csv` as well as in `scenario_config_coupled.csv`. This is what I did based on your input from this morning. 
As suggested in the Mattermost conversation, I delete `scenario_config_SSP3.csv` from the develop branch as all settings are now supposed to be in `scenario_config.csv` and the `scenario_config_SSP3.csv` is outdated in terms of several switches. The idea is to have one "ground truth" instead of two files that will likely not stay synced. 
For specific testing and experimentations, I usually work with my own config files on a local branch (which is what most people do, I think).

**Comment by nicobauer** [comment]
OK. It was the "Maybe I missed something along the way" on my Bingo-card. Thus I will approve.

**Review by nicobauer — APPROVED** [review]

**Comment by flohump** [comment]
> @flohump : Should be set `cfg_mag$gms$s15_elastic_demand <- 1` also in our default coupling config file? Or was this a ScenarioMIP specific thing that should not be used as the default? Thanks!

Yes, can be set as a default.

**Comment by laurinks** [comment]
FYI @dklein-pik: I needed to add `SSP2-NPi` to the coupled config since this scenario is used as BAU reference in NDC scenarios.

**Pull request merged**